text/0000-networkobject-parenting.md

```diff
@@ -0,0 +1,222 @@
+- Feature Name: `networkobject-parenting`
+- Start Date: 2021-05-26
+- RFC PR: [Unity-Technologies/com.unity.multiplayer.rfcs#0000](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/pull/0000)
+- Issue: [Unity-Technologies/com.unity.multiplayer#0000](https://github.com/Unity-Technologies/com.unity.multiplayer/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes a way to reparent `NetworkObject`s to other `NetworkObject`s at runtime while networking.
+
+The approach we are presenting here is limited by current MLAPI architecture and its internal structure. We will touch on these limitations and reasons below.
+
+# Motivation
+[motivation]: #motivation
+
+We want to offer `NetworkObject` reparenting solution within MLAPI to help developers synchronizing `transform` parent-child relationships of `NetworkObject`s.
+
+It is often not easy to grok the parenting in networking, also your choice of networking framework would have its limitations you don't necessarily know up until trying to implement and cover the edge cases. Here, we want to provide a solution that plays nicely with current MLAPI limitations and we expect this "NetworkObject Parenting" concept to evolve and change over time when MLAPI matures.
+
+With this RFC implemented, we expect developers to reparent their `NetworkObject`s at runtime during networked gameplay and synchronize across the network (server to clients) out-of-the-box within certain limits and conditions.
+
+# Guide-level explanation
```

Comment:
I'm going to inline Luke's concern we discussed recently here and address it later.
We want to have an escape hatch to skip this feature entirely for expert netcode devs who know what they are doing. My immediate thinking is to offer that escape hatch via a flag on NetworkObject and let devs opt-out from auto-networkobject-parenting. I'll inline that solution and give a little more details soon.

Reply:
done.